### DISCUSSION TITLE
Correct way to retry messages

### DISCUSSION THREAD

**Discussion by kibertoad** [bbc/sqs-consumer#408] [Help]
### Summary

Documentation says this:
```
Throwing an error (or returning a rejected promise) from the handler function will cause the message to be left on the queue. An [SQS redrive policy](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/SQSDeadLetterQueue.html) can be used to move messages that cannot be processed to a dead letter queue.
```

However, that's not the behaviour that I'm seeing, at least not with the default SQS configuration.

This is the structure of the message that I'm sending:
```
      const input = { 
        QueueUrl: 'some-queue-name',
        MessageBody: JSON.stringify(message),
      } satisfies SendMessageCommandInput;
```

Consumer is created like this:
```
    this.consumer = Consumer.create({
      queueUrl: 'some-queue-name',
      handleMessage: async (message: SQSMessage) => {

        throw new Error('dummy')
}})

this.consumer.start()
```

When I'm running a test with this code, I see that consumer handler is only fired once, so message is not retried.
Am I missing something?

I'm happy to expand existing documentation with an example after I understand how it works :)

**Comment by kibertoad** [comment]
Update: this was related to VisibilityTimeout.

Would a PR mentioning that this is an important thing to consider be useful in this repo?

**Comment by nicholasgriffintn** [comment]
Hey, not sure exactly what you mean by visibility timeout but we're always happy to take a contribution, especially to improve the docs :)
